lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -69,16 +69,23 @@ def include_to_hash(value)
           when String, Symbol
             {value => {}}
           when Array
-            value.flatten.each_with_object({}) { |k, h| h[k] = {} }
+            value.flatten.reduce({}) do |h, k|
+              if k.kind_of?(Hash)
+                merge_includes(h, k)
+              else
+                h[k] ||= {}
+                h
+              end
+            end
```

Comment:
Super-minor, but I'm thikning maybe we shouldn't rely on merge_includes returning the hash?  Then we can DRY up the "return the h hash" part

```suggestion
            value.flatten.reduce({}) do |h, k|
              if k.kind_of?(Hash)
                merge_includes(h, k)
              else
                h[k] ||= {}
              end
              h
            end
```

Reply:
made that change. it squashed it down even further